### PR TITLE
refactor(Balance): AND-1216 Use BalanceReporter adapters and more use…

### PR DESCRIPTION
**Pull request by westonal**
… of CryptoValue over Long/BigInteger/BigDecimal

### CODE REVIEW

File: balance/src/main/kotlin/info/blockchain/balance/CryptoValue.kt [hunk 1]
```diff
@@ -1,6 +1,38 @@
...
+
 data class CryptoValue(
     val currency: CryptoCurrency,
-    val amount: Long
-)
+
+    /**
+     * Amount in the smallest unit of the currency, Satoshi/Wei for example.
+     */
+    val amount: BigInteger
```
Comment (westonal) [inline balance/src/main/kotlin/info/blockchain/balance/CryptoValue.kt]: `Long` isn't enough for Ether. And we had `BigInteger` in Bitcoin/BitcoinCash places anyway.

File: testutils/src/main/java/piuk/blockchain/android/testutils/RxTest.kt [hunk 2]
```diff
@@ -9,8 +9,9 @@ import org.junit.Rule
  * same Scheduler that runs immediately. Also exposes a [TestScheduler] for testing of
  * time-based methods.
  */
+open class RxTest
 @Deprecated("Implement an rxInit rule locally, specifying just the schedulers the test requires")
-open class RxTest {
+constructor() {
```
Comment (westonal) [inline testutils/src/main/java/piuk/blockchain/android/testutils/RxTest.kt]: By deprecating the constructor I halved the number of warnings, we were getting one for imports as well I noticed.

File: app/src/test/java/piuk/blockchain/android/ui/dashboard/DashboardPresenterTest.kt [hunk 3]
```diff
@@ -569,8 +566,7 @@ class DashboardPresenterTest : RxTest() {
         verify(ethDataManager).fetchEthAddress()
         verify(payloadDataManager).updateAllBalances()
         verify(payloadDataManager).updateAllTransactions()
-        verify(transactionListDataManager).getBtcBalance(any())
-        verify(transactionListDataManager).getBchBalance(any())
+        verifyBalanceQueriesForBtcAndBch()
```
Comment (westonal) [inline app/src/test/java/piuk/blockchain/android/ui/dashboard/DashboardPresenterTest.kt]: I think this test over-verifies. There is usually no need to both setup the mock response and verify it was called but I've kept both for now.
It's so hard to change the presenter due to the tests, test driving a new Presenter class from scratch would be my recommendation next time we need to do significant changes to this class.

Comment (ditn) [reply app/src/test/java/piuk/blockchain/android/ui/dashboard/DashboardPresenterTest.kt]: Yeah I agree, the tests for this class have been very brittle for some time. Needs a rethink.

**Pull request merged**